Library/Homebrew/cmd/leaves.rb

```diff
@@ -15,14 +15,40 @@ def leaves_args
       description <<~EOS
         List installed formulae that are not dependencies of another installed formula.
       EOS
+      switch "-r", "--installed-on-request",
+             description: "Only list leaves that were manually installed."
+      switch "-p", "--installed-as-dependency",
```

Comment:
Why `-p`?

Reply:
`-d` is taken by debug. `-p` is for de***p***endency. If you can think of something preferable, I'll change it. I can also remove if you think it's too confusing and can't think of anything better.

Reply:
Yea, `-p` is fine 👍

Reply:
I'd be tempted by just having the long flags, I think the short ones are a little confusing here.

Reply:
>I'd be tempted by just having the long flags, I think the short ones are a little confusing here.
I agree that they're a little confusing, but the long flags are quite long. I can't imagine many more flags being added to the `leaves` command in the future. I don't think they'd be doing any harm.

Reply:
You want to take a look at `conflicts`, like in:
https://github.com/Homebrew/brew/blob/a92c194bfbd6763e4f8a0d02f1135c0f646b5aa5/Library/Homebrew/cmd/list.rb#L57